### PR TITLE
Add GitLab plugin

### PR DESCRIPTION
**Pull request by dcramer**
Requires getsentry/sentry#4081

---
This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry-plugins/11)

### CODE REVIEW

**Comment by dcramer** [comment]
Actually this needs to handle the "link existing" bit yet

**Comment by dcramer** [comment]
This is done -- can't reasonably handle the link flow due to limitations of GitLab's API.

File: src/sentry_plugins/gitlab/endpoints.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#         repo = self.get_option('repo', group.project)
+#         client = self.plugin.get_client(group.project)
+#         response = client.search_issues(repo, query)
+
+#         issues = [{
+#             'text': '(#%s) %s' % (i['number'], i['title']),
+#             'id': i['number']
+#         } for i in response.get('items', [])]
+
+#         return Response({field: issues})
```
Comment (macqueen) [inline src/sentry_plugins/gitlab/endpoints.py]: sad :(

Comment (dcramer) [inline src/sentry_plugins/gitlab/endpoints.py]: I was going to leave it as I *think* GitLab is going to get in some APIs we need, and it's a good example bit we can pull from moving forward.

**Pull request merged**